src/flows/helpers/payloads.py

```diff
@@ -73,10 +74,11 @@ def build_room_state_payload(caller: BaseState, room: BaseState) -> Dict[str, An
         else:
             objects.append(serialized)
 
-    try:
-        active_scene = cast(Any, room).active_scene
-    except AttributeError:
-        active_scene = None
+    active_scene = None
+    if isinstance(room, ActiveSceneCapable):
+        active_scene = room.active_scene
+    elif hasattr(room, 'active_scene'):
+        active_scene = room.active_scene
```

Comment:
I'm really not a fan of isinstance and hasattr checks. I'd prefer to add `active_scene` as a property/class variable in BaseState to remove the need for this.